### PR TITLE
Clean up port injection

### PR DESCRIPTION
**Pull request by davidfowl**
- We inject ASPNETCORE_URLS, HTTPS_PORT and a PORTS env variable for binding.

### CODE REVIEW

File: src/Microsoft.Tye.Core/CombineStep.cs [hunk 1]
```diff
@@ -28,6 +28,7 @@ public override Task ExecuteAsync(OutputContext output, Application application,
                     var port = binding.Port ?? 80;
                     var urls = $"http://*{(port == 80 ? "" : (":" + port.ToString()))}";
                     service.Service.Environment.Add("ASPNETCORE_URLS", urls);
+                    service.Service.Environment.Add("PORT", port);
```
Comment (rynowak) [inline src/Microsoft.Tye.Core/CombineStep.cs]: Nice. Is the idea that this is the way to get the listening port for any non-asp.net tech?

File: src/Microsoft.Tye.Hosting/ProxyService.cs [hunk 2]
```diff
@@ -69,6 +69,15 @@ static int GetNextPort()
                                         binding.Port = GetNextPort();
                                     }
 
+                                    if (binding.Protocol == "http" || binding.Protocol == null)
+                                    {
+                                        binding.InternalPort = 80;
+                                    }
+                                    else if (binding.Protocol == "https")
+                                    {
+                                        binding.InternalPort = 443;
```
Comment (rynowak) [inline src/Microsoft.Tye.Hosting/ProxyService.cs]: Can you add internal port to the docs when you get a chance? I'm not 💯 on what it does, and so it's not there.

File: src/Microsoft.Tye.Core/CombineStep.cs (hunk 1 above)
Comment (davidfowl) [reply src/Microsoft.Tye.Core/CombineStep.cs]: Yap

File: src/Microsoft.Tye.Hosting/ProxyService.cs (hunk 2 above)
Comment (davidfowl) [reply src/Microsoft.Tye.Hosting/ProxyService.cs]: It’s possible the same is bad as well. It’s really the container port.

**Pull request merged**